htdocs/expensereport/class/expensereport.class.php

```diff
@@ -1065,6 +1065,36 @@ public function delete(User $fuser = null)
             $resql=$this->db->query($sql);
             if ($resql)
             {
+            	// On efface le repertoire de pdf provisoire
+            	$ref = dol_sanitizeFileName($this->ref);
+            	if ($conf->expensereport->dir_output && !empty($this->ref))
+            	{
+            		$dir = $conf->expensereport->dir_output . "/" . $ref;
+            		$file = $conf->expensereport->dir_output . "/" . $ref . "/" . $ref . ".pdf";
+            		if (file_exists($file))	// We must delete all files before deleting directory
+            		{
+            			$ret=dol_delete_preview($this);
+
+            			if (! dol_delete_file($file, 0, 0, 0, $this)) // For triggers
```

Comment:
Do we really need to delete the file here ? Just after we delete all dir and content so we should be able to avoid this. Am i wrong ?